File: ios/Classes/BTWebViewFlutterPlugin.m
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#import "BTWebViewFlutterPlugin.h"
#import "BTCookieManager.h"
#import "FlutterWebView.h"

@implementation BTWebViewFlutterPlugin

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  BTWebViewFactory* webviewFactory =
      [[BTWebViewFactory alloc] initWithMessenger:registrar.messenger];
  [registrar registerViewFactory:webviewFactory withId:@"kr.co.bootpay/webview"];
  [BTCookieManager registerWithRegistrar:registrar];
}

@end

File: macos/Classes/BTWebViewFlutterPlugin.swift
//
//  BTWebViewFlutterPlugin.swift
//  bootpay_webview_flutter
//
//  Created by Taesup Yoon on 2021/07/06.
//

import Flutter

public class BTWebViewFlutterPlugin: NSObject, FlutterPlugin {
   public static func register(with registrar: FlutterPluginRegistrar) {
       let webviewFactory = BTWebViewFactory(messenger: registrar.messenger())
       registrar.register(webviewFactory, withId: "kr.co.bootpay/webview")
       BTCookieManager.register(with: registrar)
   } 
}
